pkg/notifications/notification_test.go

```diff
@@ -0,0 +1,83 @@
+// SPDX-FileCopyrightText: 2024 Greenbone AG <https://greenbone.net>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+package notifications
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"sync/atomic"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClient_CreateNotification(t *testing.T) {
+	tests := []struct {
+		name                   string
+		serverHttpResponseCode int
+		wantErr                bool
+	}{
+		{
+			name:                   "notification can be sent",
+			serverHttpResponseCode: http.StatusOK,
+			wantErr:                false,
+		},
+		{
+			name:                   "client returns error on notification service error",
+			serverHttpResponseCode: http.StatusInternalServerError,
+			wantErr:                true,
+		},
+	}
```

Comment:
you might also want to add a test case for "connection refused". This might not fit into this test function though, so may we need another one for this. You could also consider testing the exponential backoff in case you should decide to implement it.

Reply:
I added a test case for temporary failure to this function. As mentioned in the other comment the policy and backof from https://github.com/hashicorp/go-retryablehttp/tree/main are used. As the they are well tested, here we just verify that we use them correctly by checking if the retry works on some retryable error (here `429 Too Many Requests`).